system/jlib/jthread.cpp

```diff
@@ -2117,9 +2154,29 @@ protected: friend class PipeWriterThread;
             close(outpipe[1]);
         if (haserror)
             close(errpipe[1]);
+        if (wantsSecrets)
+            close(secretpipe[0]);
         hInput = hasinput?inpipe[1]:((HANDLE)-1);
         hOutput = hasoutput?outpipe[0]:((HANDLE)-1);
         hError = haserror?errpipe[0]:((HANDLE)-1);
+        if (wantsSecrets)
+        {
+            close(secretpipe[0]);
```

Comment:
didnt a close(secretpipe[0]) just happen above ?

Reply:
Yes - will remove one of them.